tests/tests/tier0/bluechi-is-online-system/test_bluechi_is_online_system.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+import logging
+from typing import Dict
+
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+
+LOGGER = logging.getLogger(__name__)
+
+AGENT_ONE = "agent-one"
+AGENT_TWO = "agent-two"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    agent_one = nodes[AGENT_ONE]
+    agent_two = nodes[AGENT_TWO]
+
+    # Test 1: Both agents and controller are online
+    LOGGER.debug("Testing with both agents and controller online.")
+    result, output = ctrl.exec_run("bluechi-is-online system")
```

Comment:
This should be a method on the `BluechiIsOnline` class, e.g. `system_is_online`. You can have a look at [BluechiCtl's list_units function](https://github.com/eclipse-bluechi/bluechi/blob/main/tests/bluechi_test/bluechictl.py#L42-L44) for reference. Here parameters are passed in for the implicit result check - so you'd pass in the expected result to be 1 for cases you'd expect bluechi-is-online to return 1 (i.e. at least one node not connected)

Reply:
Thanks :smiley: